castai/resource_workload_scaling_policy.go

```diff
@@ -437,6 +482,38 @@ func validateResourcePolicy(r sdk.WorkloadoptimizationV1ResourcePolicies, res st
 	if err != nil {
 		return fmt.Errorf("field %q: %w", res, err)
 	}
+
+	err = validateResourceApplyThresholdStrategy(r.ApplyThresholdStrategy)
+	if err != nil {
+		return fmt.Errorf("field %q: %w", res, err)
+	}
+	return nil
+}
+
+func validateResourceApplyThresholdStrategy(r *sdk.WorkloadoptimizationV1ApplyThresholdStrategy) error {
+	if r == nil {
+		return nil
+	}
+
+	if r.PercentageThreshold != nil {
+		if err := validatePercentageThresholdStrategy(r.PercentageThreshold); err != nil {
+			return fmt.Errorf(`field %q: %w`, FieldApplyThresholdStrategy, err)
+		}
+		return nil
+	}
+
+	return fmt.Errorf(`field %q: field %q: unknown apply threshold strategy type`, FieldApplyThresholdStrategy, FieldApplyThresholdStrategyType)
+}
+
+func validatePercentageThresholdStrategy(r *sdk.WorkloadoptimizationV1ApplyThresholdStrategyPercentageThreshold) error {
+	if r.Percentage == 0 {
+		return fmt.Errorf(
+			`field %q: field %q: value must be set for strategy type %s`,
+			FieldApplyThresholdStrategy,
+			FieldApplyThresholdStrategyPercentage,
+			FieldApplyThresholdStrategyPercentageType,
+		)
```

Comment:
```suggestion
		return fmt.Errorf(
			`field %q: value must be set for strategy type %s`,
			FieldApplyThresholdStrategyPercentage,
			FieldApplyThresholdStrategyPercentageType,
		)
```

it's handled in the above layer, so not it's duplicated: `field "cpu": field "apply_threshold_strategy": field "apply_threshold_strategy": field "percentage": value must be set for strategy type PERCENTAGE`